File: MATLAB/Spectrum_Absorption.m
%{
Created on Fri Oct 20 19:24:24 2017
@author: Lihua Xu  (libby)
%}

clc
close all %%Close all the plot, command "close all"----------------------%%
clear all
clc

%Scalling_factor=1000000;

%%load 9 files data to the newly made matrix-----------------------------%%
%%-----------------------------------------------------------------------%%
format long
load muxx.dat
length_step=length(muxx);
load muyx.dat
load muzx.dat
load muxy.dat
load muyy.dat
load muzy.dat
load muxz.dat
load muyz.dat
load muzz.dat
MuXx=muxx;
MuXy=muyx;
MuXz=muzx;
MuYx=muxy;
MuYy=muyy;
MuYz=muzy;
MuZx=muxz;
MuZy=muyz;
MuZz=muzz;

%%Shift the first value to 0 for each file (total 9)---------------------%%
%%-----------------------------------------------------------------------%%
Fisrt_value_Xx=MuXx(1,2);
Fisrt_value_Xy=MuXy(1,2);
Fisrt_value_Xz=MuXz(1,2);
Fisrt_value_Yx=MuYx(1,2);
Fisrt_value_Yy=MuYy(1,2);
Fisrt_value_Yz=MuYz(1,2);
Fisrt_value_Zx=MuZx(1,2);
Fisrt_value_Zy=MuZy(1,2);
Fisrt_value_Zz=MuZz(1,2);

MuXx(:,2)=MuXx(:,2)-Fisrt_value_Xx;
MuXy(:,2)=MuXy(:,2)-Fisrt_value_Xy;
MuXz(:,2)=MuXz(:,2)-Fisrt_value_Xz;
MuYx(:,2)=MuYx(:,2)-Fisrt_value_Yx;
MuYy(:,2)=MuYy(:,2)-Fisrt_value_Yy;
MuYz(:,2)=MuYz(:,2)-Fisrt_value_Yz;
MuZx(:,2)=MuZx(:,2)-Fisrt_value_Zx;
MuZy(:,2)=MuZy(:,2)-Fisrt_value_Zy;
MuZz(:,2)=MuZz(:,2)-Fisrt_value_Zz;

%%Average xx yy zz values from the files---------------------------------%%
%%-----------------------------------------------------------------------%%
Average_xxyyzz(:,1)=MuXx(:,1);
Average_xxyyzz(:,2)=(MuXx(:,2)+MuYy(:,2)+MuZz(:,2))./3;
figure(1)
set(gcf,'unit','normalized','position',[0.2,0.2,0.36,0.4])
plot(Average_xxyyzz(:,1),Average_xxyyzz(:,2),'-','Linewidth',1)
set(gca,'FontWeight','bold','Linewidth',1.3,'ticklength',2*get(gca,'ticklength'))
%axis auto
xlabel('Time (fs)','FontWeight','bold')
ylabel('Dipole moment','FontWeight','bold')
title('Before applying damping function','Fontsize',13)
%print('Before_applying_damping_function','-dpng','-r600');

%%Applying damping factor------------------------------------------------%%
%%-----------------------------------------------------------------------%%
dt=0.2; %atomic points
damp=zeros(length_step,1);
pdamp=0.008;
i=1:length_step;
j=i-1;
damp(i)=exp(-j.*dt*pdamp);
Average_xxyyzz(:,2)=Average_xxyyzz(:,2).*damp;
figure(2)
set(gcf,'unit','normalized','position',[0.2,0.2,0.36,0.4])
plot(Average_xxyyzz(:,1),Average_xxyyzz(:,2))
set(gca,'FontWeight','bold','Linewidth',1.3,'ticklength',2*get(gca,'ticklength'))
xlabel('Time (fs)','FontWeight','bold')
ylabel('Dipole moment','FontWeight','bold')
title('After applying damping function','Fontsize',13)
%print('After_applying_damping_function','-dpng','-r600');
%%You could add more zeros (8000 or more lines) at the end of the
%%Average_xxyyzz, This could help to increase the accuracy of the FTT
%%following.

%%Fourier transform------------------------------------------------------%%
%%-----------------------------------------------------------------------%%
%%Adding more data points into the data set (value equal to 0)-----------%%
temp=zeros(62013,2);
M=length(temp);
number_temp=length_step+1:length_step+M;
N=length_step+M;
temp(:,1)=(number_temp-1)*(MuXx(2,1)-MuXx(1,1));
Average_xxyyzz=[Average_xxyyzz;temp];
Average_xxyyzz_fft = fft(Average_xxyyzz(:,2),N);
imaginary_part=imag(Average_xxyyzz_fft);
imaginary_part=imaginary_part*(-1);
%length(imaginary_part)
%Finding what happened to the imaginary part
figure(3)
set(gcf,'unit','normalized','position',[0.2,0.2,0.36,0.4])
plot(1:length_step+M,imaginary_part)
%plot(1:length_step+M,imaginary_part.*Scalling_factor)
set(gca,'FontWeight','bold','Linewidth',1.3,'ticklength',2*get(gca,'ticklength'))
xlabel('Number','FontWeight','bold')
ylabel('Imaginary part','FontWeight','bold')
title('Temporary plot for imaginary part','Fontsize',13)
%print('Temporary_plot_for_imaginary_part','-dpng','-r600');
%%After FFT, only consider the first 5000 points, which is enough to get
%%the following results

%%Unit transfer----------------------------------------------------------%%
%%HZ---------------------------------------------------------------------%%
f=(0:(N-1))./(N*0.2); % Frequency atomic unit
Energy=f.*2*pi; % Energy atomic unit
Energy_ev=Energy.*27.2113961317875; % Energy eV unit
%length (Energy_ev)
figure(4)
set(gcf,'unit','normalized','position',[0.2,0.2,0.36,0.4])
plot(Energy_ev,imaginary_part,'Linewidth',2)
%plot(Energy_ev,imaginary_part.*Scalling_factor,'Linewidth',2)
set(gca,'FontWeight','bold','Linewidth',1.3,'ticklength',2*get(gca,'ticklength'))
xlabel('Energy (eV)','FontWeight','bold')
ylabel('Absorbance (arb.unit)','FontWeight','bold')
title('Absorbtance Spectrum','Fontsize',13)
axis([0 10 0 1.5])
%print('Absorbtance_Spectrum_Energy_Ev_Intensity','-dpng','-r600');

%%Unit transfer----------------------------------------------------------%%
%%Getting the Wavelength vs Energy------Experiment needed----------------%%
Wavelength=1239.8./Energy_ev;
figure(5)
set(gcf,'unit','normalized','position',[0.2,0.2,0.36,0.4])
plot(Wavelength,imaginary_part,'Linewidth',2)
%plot(Wavelength,imaginary_part.*Scalling_factor,'Linewidth',2)
set(gca,'FontWeight','bold','Linewidth',1.3,'ticklength',2*get(gca,'ticklength'))
xlabel('Wavelength (nm)','FontWeight','bold')
ylabel('Absorbance (arb.unit)','FontWeight','bold')
title('Absorbance Spectrum','Fontsize',13)
axis([100 1000 0 1.5])
%print('Absorbance_Spectrum_Wavelength_Intensity','-dpng','-r600');
%For Absorptance vs Energy(eV), We are interested in the energy range
%0-40eV.
%For Absorptance vs Wavelength (nm), We are interested in the wavelength
%range 100 to 1000 nm.

%%Applying the damping factor and FFT for all 9 cases--------------------%%
%%In order to find the polarizability vector-----------------------------%%
%%-----------------------------------------------------------------------%%
MuXx(:,2)=MuXx(:,2).*damp;
MuXy(:,2)=MuXy(:,2).*damp;
MuXz(:,2)=MuXz(:,2).*damp;
MuYx(:,2)=MuYx(:,2).*damp;
MuYy(:,2)=MuYy(:,2).*damp;
MuYz(:,2)=MuYz(:,2).*damp;
MuZx(:,2)=MuZx(:,2).*damp;
MuZy(:,2)=MuZy(:,2).*damp;
MuZz(:,2)=MuZz(:,2).*damp;

%%Applying the Fourier transform and FFT for all 9 cases-----------------%%
%%In order to find the polarizability vector-----------------------------%%
%%-----------------------------------------------------------------------%%
MuXx=[MuXx;temp];
MuXx_fft=fft(MuXx(:,2),N);
MuXx_imaginary_part=imag(MuXx_fft);
MuXx_imaginary_part=MuXx_imaginary_part*(-1);

MuXy=[MuXy;temp];
MuXy_fft=fft(MuXy(:,2),N);
MuXy_imaginary_part=imag(MuXy_fft);
MuXy_imaginary_part=MuXy_imaginary_part*(-1);

MuXz=[MuXz;temp];
MuXz_fft=fft(MuXz(:,2),N);
MuXz_imaginary_part=imag(MuXz_fft);
MuXz_imaginary_part=MuXz_imaginary_part*(-1);

MuYx=[MuYx;temp];
MuYx_fft=fft(MuYx(:,2),N);
MuYx_imaginary_part=imag(MuYx_fft);
MuYx_imaginary_part=MuYx_imaginary_part*(-1);

MuYy=[MuYy;temp];
MuYy_fft=fft(MuYy(:,2),N);
MuYy_imaginary_part=imag(MuYy_fft);
MuYy_imaginary_part=MuYy_imaginary_part*(-1);

MuYz=[MuYz;temp];
MuYz_fft=fft(MuYz(:,2),N);
MuYz_imaginary_part=imag(MuYz_fft);
MuYz_imaginary_part=MuYz_imaginary_part*(-1);

MuZx=[MuZx;temp];
MuZx_fft=fft(MuZx(:,2),N);
MuZx_imaginary_part=imag(MuZx_fft);
MuZx_imaginary_part=MuZx_imaginary_part*(-1);

MuZy=[MuZy;temp];
MuZy_fft=fft(MuZy(:,2),N);
MuZy_imaginary_part=imag(MuZy_fft);
MuZy_imaginary_part=MuZy_imaginary_part*(-1);

MuZz=[MuZz;temp];
MuZz_fft=fft(MuZz(:,2),N);
MuZz_imaginary_part=imag(MuZz_fft);
MuZz_imaginary_part=MuZz_imaginary_part*(-1);

%%Extract peak related to that in averaged value for these 9 cases-------%%
%%In order to find the polarizability vector-----------------------------%%
%%-----------------------------------------------------------------------%%
%%Find indexes for containning Energy_ev between 0 to 10-----------------%%
%%Dealing with the first 5000 points-------------------------------------%%
Energy_ev_temp=Energy_ev(1:5000);
indexes=find(Energy_ev_temp>=0&Energy_ev_temp<=10);
%%Energy we are interested are always within this range------------------%%
small_value=min(indexes);
big_value=max(indexes);
%%Find indexes for the peak for averged data among xx yy zz--------------%%
%max(imaginary_part(small_value:big_value))
max_peak_index_Ave=find(imaginary_part==max(imaginary_part(small_value:big_value)));
%%Disp the Energy_ev for the max_peak_index_Ave
disp('The Energy (eV) for the max peak during this period is')
Energy_ev(max_peak_index_Ave)
%%The range to find the largest peak (intensity)


%%This is using the lower triangle as the base
matrix_nine=zeros(3,3);
matrix_nine(2,1)=MuYx_imaginary_part(max_peak_index_Ave);
matrix_nine(3,1)=MuZx_imaginary_part(max_peak_index_Ave);
matrix_nine(3,2)=MuZy_imaginary_part(max_peak_index_Ave);
matrix_nine(1,1)=MuXx_imaginary_part(max_peak_index_Ave);
matrix_nine(2,2)=MuYy_imaginary_part(max_peak_index_Ave);
matrix_nine(3,3)=MuZz_imaginary_part(max_peak_index_Ave);
matrix_nine(1,2)=matrix_nine(2,1);
matrix_nine(2,3)=matrix_nine(3,2);
matrix_nine(1,3)=matrix_nine(3,1);
%matrix_nine

%{ 
%%This is using the upper triangle as the base
matrix_nine(1,1)=MuXx_imaginary_part(max_peak_index_Ave);
matrix_nine(1,2)=MuXy_imaginary_part(max_peak_index_Ave);
matrix_nine(1,3)=MuXz_imaginary_part(max_peak_index_Ave);
matrix_nine(2,1)=matrix_nine(1,2);
%matrix_nine(2,1)=MuYx_imaginary_part(max_peak_index_Ave);
matrix_nine(2,2)=MuYy_imaginary_part(max_peak_index_Ave);
matrix_nine(2,3)=MuYz_imaginary_part(max_peak_index_Ave);
matrix_nine(3,1)=matrix_nine(1,3);
%matrix_nine(3,1)=MuZx_imaginary_part(max_peak_index_Ave);
matrix_nine(3,2)=matrix_nine(2,3);
%matrix_nine(3,2)=MuZy_imaginary_part(max_peak_index_Ave);
matrix_nine(3,3)=MuZz_imaginary_part(max_peak_index_Ave);
%}

%%Getting the eigenvalues and eigenvectors for the above matrix----------%%
%%In order to find the polarizability vector-----------------------------%%
%%-----------------------------------------------------------------------%%
[Eigenvectors,Eigenvalues] = eig(matrix_nine);
%%Finding the maxmum eigenvalue
value_in_matrix=max(max(Eigenvalues));
[row col]=find(value_in_matrix==Eigenvalues);
disp('1-x, 2-y,3-z,so the largest number for the eigenvalues is')
row
disp('The conrresponding eigenvectors is')
Eigenvectors(:,row)

%close all 
%%clear all
%%clc

File: Python/Spectrum_Absorption.py
"""
Created on Thu Oct 19 16:21:33 2017
@author: Lihua Xu (libby)
"""

#Scalling_factor=1000000;

##load 9 files data to the newly made matrix
import numpy as np
MuXx = np.genfromtxt('muxx.dat',delimiter='')
MuXy = np.genfromtxt('muyx.dat',delimiter='')
MuXz = np.genfromtxt('muzx.dat',delimiter='')
MuYx = np.genfromtxt('muxy.dat',delimiter='')
MuYy = np.genfromtxt('muyy.dat',delimiter='')
MuYz = np.genfromtxt('muzy.dat',delimiter='')
MuZx = np.genfromtxt('muxz.dat',delimiter='')
MuZy = np.genfromtxt('muyz.dat',delimiter='')
MuZz = np.genfromtxt('muzz.dat',delimiter='')

#print(MuXx)
#print(MuXy)
#print(MuXz)
#print(MuYx)
#print(MuYy)
#print(MuYz)
#print(MuZx)
#print(MuZy)
#print(MuZz)

Fisrt_value_Xx=MuXx[0][1]
Fisrt_value_Xy=MuXy[0][1]
Fisrt_value_Xz=MuXz[0][1]
Fisrt_value_Yx=MuYx[0][1]
Fisrt_value_Yy=MuYy[0][1]
Fisrt_value_Yz=MuYz[0][1]
Fisrt_value_Zx=MuZx[0][1]
Fisrt_value_Zy=MuZy[0][1]
Fisrt_value_Zz=MuZz[0][1]

##Shift the second column in order to make the first value be 0

MuXx_mu=MuXx[:,1]-Fisrt_value_Xx
MuXy_mu=MuXy[:,1]-Fisrt_value_Xy
MuXz_mu=MuXz[:,1]-Fisrt_value_Xz
MuYx_mu=MuYx[:,1]-Fisrt_value_Yx
MuYy_mu=MuYy[:,1]-Fisrt_value_Yy
MuYz_mu=MuYz[:,1]-Fisrt_value_Yz
MuZx_mu=MuZx[:,1]-Fisrt_value_Zx
MuZy_mu=MuZy[:,1]-Fisrt_value_Zy
MuZz_mu=MuZz[:,1]-Fisrt_value_Zz
length_step=len(MuZz_mu)
Mu_num=np.zeros((length_step,1))
for t in range(0,length_step):
    Mu_num[t]=MuZz[t][0]

#Average xx yy zz values from the files
Average_xxyyzz=(MuXx_mu+MuYy_mu+MuZz_mu)/3
import matplotlib.pylab as pl
pl.figure(1)
#pl.subplot(311)
pl.plot(Mu_num, Average_xxyyzz)
pl.title('Before applying damping function')# give plot a title
pl.xlabel('Time (fs)')# make axis labels
pl.ylabel('Dipole moment')
pl.xlim(0, 100)# set axis limits
#pl.ylim(0, 1)
pl.show()

#Applying damping factor
dt=0.2
pdamp=0.008
damp=np.zeros((length_step,1))
Average_xxyyzz_new=np.zeros((length_step,1))
for i in range(0,length_step):
    damp[i]=np.exp(-i*dt*pdamp)
for j in range(0,length_step):
    Average_xxyyzz_new[j]=Average_xxyyzz[j]*damp[j]
pl.figure(2)
#pl.subplot(312)
pl.plot(Mu_num, Average_xxyyzz_new)
pl.title('After applying damping function')# give plot a title
pl.xlabel('Time (fs)')# make axis labels
pl.ylabel('Dipole moment')
pl.xlim(0, 100)# set axis limits    
#pl.ylim(0, 1)
pl.show()

#Fourier transform
temp_mu=np.zeros(((length_step-1)*3,1))
temp_N=len(temp_mu)
Average_extended=np.vstack((Average_xxyyzz_new,temp_mu))
N=temp_N+length_step
temp_num=np.zeros(((length_step-1)*3,1))
for mm in range(length_step,N):
    temp_num[mm-length_step]=mm*(Mu_num[1]-Mu_num[0])
Mu_extended=np.vstack((Mu_num,temp_num))
##Do Fourier transform
from scipy.fftpack import fft
Average_extended_new=Average_extended[:,0]
Average_extended_fft=fft(Average_extended_new)
Average_imaginary = Average_extended_fft.imag
Average_imaginary = Average_imaginary*(-1)
order=np.zeros((N,1))
for ni in range(0,N):
    order[ni]=ni  
pl.figure(3)
#pl.subplot(313)
pl.plot(order, Average_imaginary)
pl.title('Temporary plot for imaginary part')# give plot a title
pl.xlabel('Number')# make axis labels
pl.ylabel('Imaginary part')
pl.xlim(0, N)# set axis limits 
pl.show()

#Unit transfer 
##HZ
import math
f=np.zeros((N,1))
Energy=np.zeros((N,1))
for Ut in range(0,N):
    f[Ut]=Ut/(dt*N)
    Energy[Ut]=f[Ut]*2*math.pi
Hartree_ev=27.2113961317875
Energy_ev=Energy*Hartree_ev
pl.figure(4)
pl.plot(Energy_ev, Average_imaginary)
pl.title('Absorbtance Spectrum')# give plot a title
pl.xlabel('Energy (eV)')# make axis labels
pl.ylabel('Absorbance (arb.unit)')
pl.xlim(0, 10)# set axis limits    
#pl.ylim(0, 1)
pl.show()

##Getting the Wavelength vs Energy  (Experiment needed)
Wavelength=1239.8/Energy_ev
pl.figure(5)
pl.plot(Wavelength, Average_imaginary)
pl.title('Absorbtance Spectrum')# give plot a title
pl.xlabel('Wavelength (nm)')# make axis labels
pl.ylabel('Absorbance (arb.unit)')
pl.xlim(100, 1000)# set axis limits    
#pl.ylim(0, 1)
pl.show()

#Applying the damping factor and FFT for all 9 cases
##In order to find the polarizability vector
MuXx_mu_damp=np.zeros((length_step,1))
MuXy_mu_damp=np.zeros((length_step,1))
MuXz_mu_damp=np.zeros((length_step,1))
MuYx_mu_damp=np.zeros((length_step,1))
MuYy_mu_damp=np.zeros((length_step,1))
MuYz_mu_damp=np.zeros((length_step,1))
MuZx_mu_damp=np.zeros((length_step,1))
MuZy_mu_damp=np.zeros((length_step,1))
MuZz_mu_damp=np.zeros((length_step,1))
for jj in range(0,length_step):
    MuXx_mu_damp[jj]=MuXx_mu[jj]*damp[jj]
    MuXy_mu_damp[jj]=MuXy_mu[jj]*damp[jj]
    MuXz_mu_damp[jj]=MuXz_mu[jj]*damp[jj]
    MuYx_mu_damp[jj]=MuYx_mu[jj]*damp[jj]
    MuYy_mu_damp[jj]=MuYy_mu[jj]*damp[jj]
    MuYz_mu_damp[jj]=MuYz_mu[jj]*damp[jj]
    MuZx_mu_damp[jj]=MuZx_mu[jj]*damp[jj]
    MuZy_mu_damp[jj]=MuZy_mu[jj]*damp[jj]
    MuZz_mu_damp[jj]=MuZz_mu[jj]*damp[jj]
    
##In order to find the polarizability vector
MuXx_extended=np.vstack((MuXx_mu_damp,temp_mu))
MuXy_extended=np.vstack((MuXy_mu_damp,temp_mu))
MuXz_extended=np.vstack((MuXz_mu_damp,temp_mu))
MuYx_extended=np.vstack((MuYx_mu_damp,temp_mu))
MuYy_extended=np.vstack((MuYy_mu_damp,temp_mu))
MuYz_extended=np.vstack((MuYz_mu_damp,temp_mu))
MuZx_extended=np.vstack((MuZx_mu_damp,temp_mu))
MuZy_extended=np.vstack((MuZy_mu_damp,temp_mu))
MuZz_extended=np.vstack((MuZz_mu_damp,temp_mu))

MuXx_extended_new=MuXx_extended[:,0]
MuXx_extended_fft=fft(MuXx_extended_new)
MuXx_imaginary = MuXx_extended_fft.imag
MuXx_imaginary = MuXx_imaginary*(-1)

MuXy_extended_new=MuXy_extended[:,0]
MuXy_extended_fft=fft(MuXy_extended_new)
MuXy_imaginary = MuXy_extended_fft.imag
MuXy_imaginary = MuXy_imaginary*(-1)

MuXz_extended_new=MuXz_extended[:,0]
MuXz_extended_fft=fft(MuXz_extended_new)
MuXz_imaginary = MuXz_extended_fft.imag
MuXz_imaginary = MuXz_imaginary*(-1)

MuYx_extended_new=MuYx_extended[:,0]
MuYx_extended_fft=fft(MuYx_extended_new)
MuYx_imaginary = MuYx_extended_fft.imag
MuYx_imaginary = MuYx_imaginary*(-1)

MuYy_extended_new=MuYy_extended[:,0]
MuYy_extended_fft=fft(MuYy_extended_new)
MuYy_imaginary = MuYy_extended_fft.imag
MuYy_imaginary = MuYy_imaginary*(-1)

MuYz_extended_new=MuYz_extended[:,0]
MuYz_extended_fft=fft(MuYz_extended_new)
MuYz_imaginary = MuYz_extended_fft.imag
MuYz_imaginary = MuYz_imaginary*(-1)

MuZx_extended_new=MuZx_extended[:,0]
MuZx_extended_fft=fft(MuZx_extended_new)
MuZx_imaginary = MuZx_extended_fft.imag
MuZx_imaginary = MuZx_imaginary*(-1)

MuZy_extended_new=MuZy_extended[:,0]
MuZy_extended_fft=fft(MuZy_extended_new)
MuZy_imaginary = MuZy_extended_fft.imag
MuZy_imaginary = MuZy_imaginary*(-1)

MuZz_extended_new=MuZz_extended[:,0]
MuZz_extended_fft=fft(MuZz_extended_new)
MuZz_imaginary = MuZz_extended_fft.imag
MuZz_imaginary = MuZz_imaginary*(-1)

#Extract peak related to that in averaged value for these 9 cases-
index_vector=np.zeros((5000,1))
iv=len(index_vector)
pp=0
for E_num in range(0,5000):
    if Energy_ev[E_num]>=0 and Energy_ev[E_num]<=10:
        index_vector[pp]=E_num
        pp=pp+1
import scipy
index_vector=scipy.delete(index_vector,range(pp,iv),0)
small_value=min(index_vector)
small_value=int(small_value[0])
big_value=max(index_vector)
big_value=int(big_value[0])
inteval=big_value-small_value+1
Average_imaginary_new=np.zeros((N,1))
for cc in range(0,N):
    Average_imaginary_new[cc]=Average_imaginary[cc]
existing_matrix_index=np.zeros((inteval,1))
existing_matrix_value=np.zeros((inteval,1))
for ff in range(0,inteval):
    existing_matrix_value[ff,0]=Average_imaginary_new[small_value+ff]
    existing_matrix_index[ff,0]=small_value+ff
peak=max(existing_matrix_value)
#peak=peak[0]
existing_matrix_value_list = existing_matrix_value.tolist()
peak_index=existing_matrix_index[existing_matrix_value_list.index(peak),0]
peak_index=int(peak_index)
print "The Energy (eV) for the max peak during this period is "
Peak_Energy=Energy_ev[peak_index][0]
print Peak_Energy

##This is using the lower triangle as the base
matrix_nine=np.zeros((3,3))
matrix_nine[0][0]=MuXx_imaginary[peak_index]
matrix_nine[1][0]=MuYx_imaginary[peak_index]
matrix_nine[2][0]=MuZx_imaginary[peak_index]
matrix_nine[2][1]=MuZy_imaginary[peak_index]
matrix_nine[1][1]=MuYy_imaginary[peak_index]
matrix_nine[2][2]=MuZz_imaginary[peak_index]
matrix_nine[0][1]=matrix_nine[1][0]
matrix_nine[1][2]=matrix_nine[2][1]
matrix_nine[0][2]=matrix_nine[2][0]
#print matrix_nine

#Getting the eigenvalues and eigenvectors for the above matrix
#In order to find the polarizability vector
from numpy import linalg as LA
w, v = LA.eig(np.array(matrix_nine))
Max_eigenvalue=max(w)
Eigenvalues = w.tolist()
Max_Eigvalue_index=Eigenvalues.index(Max_eigenvalue)
print "The maximum eigenvalue is "
print Eigenvalues[Max_Eigvalue_index]
print "The conrresponding eigenvectors is for x y and z direction are "
print v[:,Max_Eigvalue_index]
